cogs/tiles.py

```diff
@@ -63,10 +59,25 @@ async def tile(self, ctx: Context, *args, smalltile=False):
             obj = qindex[nearest[0]]
         if obj.tile is not None:
             tile = obj.tile
-            if recolor != '':
+            gif = None
+            msg = ''
+            if hologram:
```

Comment:
The animation calls should be broken out here into asynchronous executors since they can take seconds to complete. This would block the bot from responding to anything else while these are running.

Here's an example from blueprints.py where I did this for the 'process.extract' function which can take multiple seconds to complete:

```
        loop = asyncio.get_running_loop()
        with concurrent.futures.ThreadPoolExecutor() as pool:
            call = partial(process.extract, query, self.ids, limit=5)
            id_matches_raw = await loop.run_in_executor(pool, call)
```

functools.partial is used only to allow passing parameters into this call. When execution here hits the 'await', the bot will start processing the animation in the background and suspend execution of the current function, becoming responsive to other user commands in other cogs until the animation completes, at which point execution will return here and resume.

This is only possible here because we are already in a function marked 'async', which is itself being awaited elsewhere. The reason to use a special executor for process.extract and the tile animators is that they are not marked 'async', meaning they are synchronous (blocking) functions which cannot be normally awaited.

Reply:
Thanks for the example and explanation. I have now implemented this and confirmed that crypto can respond to subsequent simple `?tile` requests while the `?hologram` or `?animate` request is still being processed.

I ended up creating a new static method (`TilesHelper.get_bytesio_for_object`) to encapsulate the target behavior because I couldn't quite figure out how to make `functools.partial` work on an instance method (I think it needs to use `functools.partial_method` in some form but it wasn't really clear to me exactly how.) In any case, it works properly now and that didn't seem like too big of a deal. 🙂